grease-aarch32/src/Grease/Macaw/Arch/AArch32.hs

```diff
@@ -86,8 +98,30 @@ armCtx halloc mbReturnAddr stackArgSlots = do
       , _archOffsetStackPointerPostCall = pure
       , -- assumes AAPCS32 https://github.com/ARM-software/abi-aa/blob/main/aapcs32/aapcs32.rst#parameter-passing
         _archABIParams = Some.Some <$> [ARM.r0, ARM.r1, ARM.r2, ARM.r3]
+      , _archPCFixup = armArchPCFixup
       }
 
+armArchPCFixup ::
```

Comment:
Let's add a Haddock saying why something nontrivial is necessary here, and how incrementing the address by 1 takes care of that.